src/eval/run/system.rs

```diff
@@ -70,6 +70,14 @@ macro_rules! try_balance {
 }
 
 pub fn create<M: Memory + Default, P: Patch>(state: &mut State<M, P>, after_gas: Gas) -> Option<Control> {
+    do_create(state, after_gas, false)
+}
+
+pub fn create2<M: Memory + Default, P: Patch>(state: &mut State<M, P>, after_gas: Gas) -> Option<Control> {
+    do_create(state, after_gas, true)
+}
+
+fn do_create<M: Memory + Default, P: Patch>(state: &mut State<M, P>, after_gas: Gas, is_create2: bool) -> Option<Control> {
```

Comment:
Can't find any other usages of `do_create`, imho better to use 2 separate functions, especially if used once

Reply:
DRY principal. Code is used 2 times, why copy-paste it when we can abstract it out to a helper function? 
Still there is a way to improve that, will abstract-out only a necessary part.

Reply:
Ah, I get what you mean. You're right, we can use the single function and call it with boolean parameter, it's not a public API anyway.

Reply:
Done.